Fix issue #39

0xc0170 opened the pull request:
Adding pins reference to readme, plus one correction

@autopulated @bremoran @bogdanm

bogdanm commented:
+1

autopulated commented:
@bremoran I think the #inlcude "PinNames.h" is intentional? (PinNames.h forwards the include to GeneratedPinNames.h if appropriate, e.g. https://github.com/ARMmbed/mbed-hal-k64f/blob/master/mbed-hal-k64f/PinNames.h, ensuring backwards compatibility for targets that don't use config. Maybe this could use an explanation in the docs.)

I had also imagined something a little more verbose... but I suppose this technically fixes the issue.

bremoran commented:
@autopulated A more verbose version was on my to-do list for today.  Martin beat me to it ;)

0xc0170 commented:
I removed  that Generated header correction, I assumed we fixed PinNames, but as Brendan pointed out, not yet :sob:  I added a PinName section. 

What else you consider adding , more verbose - which sections? I can add it within this PR

bremoran commented on pins.md:
```diff
@@ -29,3 +29,7 @@ Pin names are an exception from typical yotta config, in that the pin names decl
+
+## PinName file
+The chip hal implementation should contain PinNames header file. This header should just include mbed-hal/GenereatedPinNames.h for backwards-compatibility.
+We would like to remove PinNames in the future, use the GeneratedPinNames header file.
```
My preference would be to remove ```PinNames.h``` and then rename ```GeneratedPinNames.h``` to ```PinNames.h```

bremoran commented:
See comment above, otherwise, +1
